bin/qunit

```diff
@@ -0,0 +1,768 @@
+    params["module"] = @module if @module
+    params["filter"] = @filter if include_filter && @filter
+    params["qunit_skip_core"] = "1" if @qunit_skip_core
+    params["qunit_single_plugin"] = @qunit_single_plugin if @qunit_single_plugin
+    params["theme_name"] = @theme_name if @theme_name
+    params["theme_url"] = @theme_url if @theme_url
+    params["theme_id"] = @theme_id if @theme_id
+    params["target"] = resolved_target if resolved_target
+    params["report_requests"] = "1" if @report_requests
+
+    build_query_string(params)
+  end
+
+  def stop_unicorn(pid, pid_path)
+    Process.kill("-KILL", pid)
+  rescue Errno::ESRCH
+  ensure
+    FileUtils.rm_f(pid_path) if pid_path
+  end
+
+  def rails_root
+    @rails_root ||= File.expand_path("..", __dir__)
+  end
+
+  def frontend_dir
+    @frontend_dir ||= File.expand_path("../frontend/discourse", __dir__)
+  end
+
+  def plugin_target_from_paths(paths)
+    plugin_names = plugin_names_from_paths(paths)
+    return nil if plugin_names.empty?
+
+    include_core = paths.any? { |path| !plugin_path?(path) }
+
+    determine_plugin_target(plugin_names, include_core: include_core)
+  end
+
+  def resolved_target
+    return @resolved_target if defined?(@resolved_target)
+
+    target = present_string(@target)
+    target ||= plugin_target_from_paths(@file_paths)
+    target ||= resolved_filter_target
+
+    @resolved_target = target
+  end
+
+  def resolved_seed
+    return @resolved_seed if defined?(@resolved_seed)
+
+    @resolved_seed = present_string(@seed) || Random.new.seed
+  end
+
+  def resolved_filter_target
+    return nil unless @filter
+
+    plugin_target_from_filter(@filter, plugin_filter?)
+  end
+
+  def plugin_filter?
+    return false unless @filter
+
+    @load_plugins || @filter.downcase.include?("plugin") || @filter.include?("plugins/")
+  end
+
+  def resolved_plugin_targets_value
+    explicit = present_string(ENV["PLUGIN_TARGETS"])
+    return explicit if explicit
+
+    return nil unless resolved_target == "plugins"
+
+    names = plugin_names_from_paths(@file_paths)
+    names = plugin_names_from_filter(@filter) if names.empty?
+
+    return nil if names.empty?
+
+    names.join(",")
+  end
+
+  def should_load_plugins?
+    return true if @load_plugins
+    return true if @file_paths.any? { |path| plugin_path?(path) }
+
+    target = resolved_target
+    target && target != "core"
+  end
+
+  def plugin_names_from_paths(paths)
+    paths.filter_map { |path| plugin_name_from_path(path) }.uniq
+  end
+
+  def plugin_name_from_path(path)
+    match = path.match(%r{(?:^|/)plugins/([^/]+)/})
+    return nil unless match
+
+    resolve_plugin_namespace(match[1])
+  end
+
+  def plugin_path?(path)
+    path.match?(%r{(?:^|/)plugins/})
+  end
+
+  def determine_plugin_target(plugin_names, include_core: false)
+    return nil if plugin_names.empty?
+    return "all" if include_core
+    return plugin_names.first if plugin_names.length == 1
+
+    "plugins"
```

Comment:
Running in this mode is not really supported. It's labelled as 'not recommended' in the UI, and causes many plugin tests to fail.

<img width="258" height="137" alt="Image" src="https://github.com/user-attachments/assets/a5845dd6-f3f2-4f31-a34c-983503687e96" />

Tests for each plugin should be run independently, with the appropriate target. Testem supports passing multiple URLs to be queued and tested in parallel. This is how it's wired up on `main` right now:

https://github.com/discourse/discourse/blob/6adb4f8e845fd0a27880b3a1f6a74413dd22014f/frontend/discourse/testem.js#L196C9-L202C1